test/delegatemulti.js

```diff
@@ -0,0 +1,164 @@
+    // const customAmount = ethers.utils.parseEther('10000000.0'); // ens
+
+    // give allowance to multi delegate contract
+    await token.approve(multiDelegate.address, delegatorTokenAmount);
+    // delegate multiple delegatees
+    const delegatees = [deployer, alice, bob, charlie];
+    await multiDelegate.delegateMulti(
+      delegatees,
+      delegatees.map((_) => delegatorTokenAmount.div(delegatees.length))
+    );
+
+    const delegatorTokenAmountAfter = await token.balanceOf(deployer);
+    expect(delegatorTokenAmountAfter.toString()).to.equal('0');
+
+
+    // delegator must have 1/2 of the votes the delegator delegated
+    const votesOfDelegator = await token.getVotes(deployer);
+    expect(votesOfDelegator.toString()).to.equal(
+      delegatorTokenAmount.div(delegatees.length).toString()
+    );
+
+    // delegatee must have 1/2 of the votes the delegator delegated
+    const votesOfDelegatee = await token.getVotes(alice);
+    expect(votesOfDelegatee.toString()).to.equal(
+      delegatorTokenAmount.div(delegatees.length).toString()
+    );
+
+    for (let delegateTokenId of delegatees) {
+      let balance = await multiDelegate.balanceOf(deployer, delegateTokenId);
+      expect(balance.toString()).to.equal(
+        delegatorTokenAmount.div(delegatees.length).toString()
+      );
+    }
+
+    await multiDelegate.withdraw(delegatees);
+
+    for (let delegateTokenId of delegatees) {
+      let balance = await multiDelegate.balanceOf(deployer, delegateTokenId);
+      expect(balance.toString()).to.equal("0");
+    }
+  });
+
+  it('contract should revert if allowance is not provided', async () => {
+    const delegatorTokenAmount = await token.balanceOf(deployer);
+
+    const delegatees = [alice];
+    await expect(
+      multiDelegate.delegateMulti(delegatees, [delegatorTokenAmount])
+    ).to.be.revertedWith('ERC20: insufficient allowance');
+  });
+
+  it('contract should revert if allowance is lesser than provided amount', async () => {
+    const delegatorTokenAmount = await token.balanceOf(deployer);
+    const customAmount = ethers.utils.parseEther('100000.0'); // total ens 77000000
+
+    // give allowance to multi delegate contract
+    await token.approve(multiDelegate.address, customAmount);
+
+    const delegatees = [bob];
+    await expect(
+      multiDelegate.delegateMulti(delegatees, [delegatorTokenAmount])
+    ).to.be.revertedWith('ERC20: insufficient allowance');
+  });
+
+  it('contract should revert if no delegatee provided', async () => {
+    const delegatorTokenAmount = await token.balanceOf(deployer);
+
+    // give allowance to multi delegate contract
+    await token.approve(multiDelegate.address, delegatorTokenAmount);
+
+    const delegatees = [];
+    await expect(
+      multiDelegate.delegateMulti(delegatees, [delegatorTokenAmount])
+    ).to.be.revertedWith('You should pick at least one delegatee');
+  });
+
+  it('contract should revert if no amount provided', async () => {
```

Comment:
There should be tests for insufficient balances too - either depositing or withdrawing.

Reply:
deposits addressed here;

https://github.com/ensdomains/governance/blob/4e4c7cbc89d82aa9f035a7abcc7a1854432eb423/test/delegatemulti.js#L619

partially here;
https://github.com/ensdomains/governance/blob/4e4c7cbc89d82aa9f035a7abcc7a1854432eb423/test/delegatemulti.js#L419

withdraw here;
https://github.com/ensdomains/governance/blob/4e4c7cbc89d82aa9f035a7abcc7a1854432eb423/test/delegatemulti.js#L533-L534